editor/components/block-list/block.js

```diff
@@ -348,6 +354,96 @@ export class BlockListBlock extends Component {
 		this.setState( { error } );
 	}
 
+	onDragStart( event ) {
+		const dragInset = document.getElementById( `block-drag-inset-${ this.props.uid }` );
+		const block = document.getElementById( `block-${ this.props.uid }` );
+
+		// Closure to remove the cloned node from the DOM (fired within timeout below)
+		const removeBlockClone = ( _blockList, _cloneWrapper ) => {
+			return () => {
+				_blockList.removeChild( _cloneWrapper );
+			};
+		};
+
+		// Closure to hide the visible block and show inset in its place (fired within timeout below)
+		const showInset = () => {
+			return () => {
+				this.setState( state => Object.assign( {}, state, { dragging: true } ) );
+				document.body.classList.add( 'dragging' );
+			};
+		};
+
+		event.dataTransfer.setData(
+			'text',
```

Comment:
If I drag an image block to my desktop I get a JSON file with this internal representation. Low-priority, but I am curious whether we can somehow affect that.

Reply:
Nice catch! I will investigate.